src/uris/deepLinks/deepLinkService.ts

```diff
@@ -119,35 +161,61 @@ export class DeepLinkService implements Disposable {
 		return undefined;
 	}
 
-	private async processDeepLink(): Promise<void> {
+	private async showOpenRepoPrompt(): Promise<OpenWorkspaceLocation | undefined> {
```

Comment:
I think we need another prompt here. This prompt is a bit confusing, since we are saying we can't find the repo, but then asking you to open it.

Instead, I think we should tell the user we couldn't find a matching repo, and ask if they want find it themselves, with buttons to "Open Folder...", "Open Workspace...", "Cancel" (if "Open Workspace..." is tricky we can punt).

Then we open the folder/file picker, and then ask where to open this folder/workspace like this prompt.

Then open the

Reply:
Done in commit: [Update verbiage, open repo flow](https://github.com/gitkraken/vscode-gitlens/pull/2503/commits/2154cca1a37764d8f4416e8b37a0ff75a990a2d0)